src/vivarium/framework/population/manager.py

```diff
@@ -490,33 +490,38 @@ def get_population(
             for attribute in attributes_to_include
             if attribute not in simple_attributes
         ]
+        contains_column_multi_index = False
         for name in remaining_attributes:
-            pipeline = self._attribute_pipelines[name]
-            values = self._population.loc[idx, name] if pipeline.is_simple else pipeline(idx)
+            values = self._attribute_pipelines[name](idx)
+
+            # Handle column names
             if isinstance(values, pd.Series):
                 if values.name is not None and values.name != name:
                     self.logger.warning(
-                        f"The '{name} attribute pipeline returned a pd.Series with a "
+                        f"The '{name}' attribute pipeline returned a pd.Series with a "
                         f"different name '{values.name}'. For the column being added to the "
                         f"population state table, we will use '{name}'."
                     )
                 values.name = name
+            else:
+                # Must be a dataframe - need to add another index level to each column
+                values.columns = pd.MultiIndex.from_product([[name], values.columns])
+                contains_column_multi_index = True
             attributes_list.append(values)
 
+        # Make sure all items of the list have consistent column levels
+        if contains_column_multi_index:
+            for i, item in enumerate(attributes_list):
+                if isinstance(item, pd.Series):
+                    item_df = item.to_frame()
+                    item_df.columns = pd.MultiIndex.from_tuples([(item.name, "")])
```

Comment:
This results in a dataframe w/ the column name on the outer level and an empty inner level. That seems to behave the closes to a single level column df.

```
>>> df
      foo         value
      bar     baz      
0  foobar  foobaz     1
1  foobar  foobaz     2
>>> df.foo
      bar     baz
0  foobar  foobaz
1  foobar  foobaz
>>> df.foo.bar
0    foobar
1    foobar
Name: bar, dtype: object
>>> df.foo.baz
0    foobaz
1    foobaz
Name: baz, dtype: object
>>> df.value
0    1
1    2
Name: value, dtype: int64
```